src/parser.rs

```diff
@@ -1017,10 +1017,31 @@ impl Parser {
             ColumnOption::Unique { is_primary: false }
         } else if self.parse_keyword("REFERENCES") {
             let foreign_table = self.parse_object_name()?;
-            let referred_columns = self.parse_parenthesized_column_list(Mandatory)?;
+            let referred_columns = self.parse_parenthesized_column_list(Optional)?;
+            let mut on_delete = None;
+            let mut on_update = None;
+            while self.parse_keyword("ON") {
+                if self.parse_keyword("DELETE") {
+                    if on_delete == None {
+                        on_delete = Some(self.parse_reference_change_action()?);
+                    } else {
+                        return self
+                            .expected("ON DELETE option not more than once", self.peek_token());
```

Comment:
I don't think this deserves a special error message, and they add some complexity. Would this work instead?
```
            let (mut on_delete, mut on_update) = (None, None);
            loop {
                if on_delete.is_none() && self.parse_keywords(vec!["ON", "DELETE"]) {
                    on_delete = Some(self.parse_reference_change_action()?);
                } else if on_update.is_none() && self.parse_keywords(vec!["ON", "UPDATE"]) {
                    on_update = Some(self.parse_reference_change_action()?);
                } else {
                    break;
                }
            }
```

Reply:
I added some tests to support that your proposal works.